impl/README.md

```diff
@@ -0,0 +1,189 @@
+![Verify JAVA SDK](https://github.com/serverlessworkflow/sdk-java/workflows/Verify%20JAVA%20SDK/badge.svg)
+![Deploy JAVA SDK](https://github.com/serverlessworkflow/sdk-java/workflows/Deploy%20JAVA%20SDK/badge.svg) [![Gitpod ready-to-code](https://img.shields.io/badge/Gitpod-ready--to--code-blue?logo=gitpod)](https://gitpod.io/#https://github.com/serverlessworkflow/sdk-java)
+
+# Serverless Workflow Specification - Java SDK- Reference Implementation
+
+Welcome to Java SDK runtime reference implementation, a lightweight implementation of the Serverless Workflow specification which provides a simple, non blocking, reactive API for workflow execution. 
+
+Although initially conceived mainly for testing purposes, it was designed to be easily expanded, so it can eventually become production ready. 
+
+## Status. 
+
+This reference implementation is currently capable of running workflows consisting of:
+
+
+* Tasks
+    * Switch 
+    * Set
+    * Do
+    * Raise
+    * Listen
+    * Emit
+    * Fork
+    * For
+    * Try
+    * Wait
+    * Call
+        * HTTP
+* Schema Validation
+    * Input
+    * Output
+* Expressions
+    * Input
+    * Output 
+    * Export
+    * Special keywords: runtime, workflow, task...
+* Error definitions
+
+
+## Setup
+
+Serverless workflow reference implementation only requires setting up Java and Maven/Gradle. 
+
+### JDK Version
+
+Reference implementation requires [Java 17](https://openjdk.org/projects/jdk/17/) or newer versions. 
+
+### Dependencies
+
+One of the goals of the reference implemetantion is to maintain the number of dependencies as lower as possible. With that spirit, a modular approach has been followed, lettting the user decide, depending on their wokflows nature, which dependencies should be include. 
+
+In practical terms, this means a separation between the core part and additional dependencies that should be explicitly included if your workflow is interacting with an external service that communicated using a particular technology supported by the specification (at this moment, just HTTP). The intention of this is to avoid adding dependencies that you do not really need (for example, when gRPC call will be implemented, if we were adding the gRPC stack to the core dependencies, you wont be able to get rid of it even if none of your workflows use it)
```

Comment:
```suggestion
In practical terms, this means a separation between the core part and additional dependencies that should be explicitly included if your workflow is interacting with an external service that communicated using a particular technology supported by the specification (at this moment, just HTTP). The intention of this is to avoid adding dependencies that you do not need (for example, when gRPC call will be implemented, if we were adding the gRPC stack to the core dependencies, you won't be able to get rid of it even if none of your workflows use it)
```